ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useWatchLastSnapshotForReports.ts

```diff
@@ -36,64 +27,57 @@ async function fetchLastReportJobForConfiguration(
+    fetchSnapshots: () => void;
 };
 
 export function useWatchLastSnapshotForReports(
     reportConfigurations: ReportConfiguration | ReportConfiguration[] | null
 ): FetchLastSnapshotReturn {
-    const [result, setResult] = useState<Result>(defaultResult);
-
-    const fetchSnapshots = useCallback(async () => {
+    const fetchSnapshotsCallback = useCallback(async () => {
         if (!reportConfigurations) {
-            return;
+            const result: ReportSnapshotLookup = {};
+            return Promise.resolve(result);
         }
 
-        setResult((prevResult) => ({
-            ...prevResult,
-            isLoading: true,
-            error: null,
-        }));
+        const promise: Promise<ReportSnapshotLookup> = new Promise((resolve, reject) => {
+            const configurations = Array.isArray(reportConfigurations)
+                ? reportConfigurations
+                : [reportConfigurations];
 
-        const configurations = Array.isArray(reportConfigurations)
-            ? reportConfigurations
-            : [reportConfigurations];
+            Promise.all(configurations.map(({ id }) => fetchLastReportJobForConfiguration(id)))
+                .then((snapshotResults) => {
+                    const result: ReportSnapshotLookup = configurations.reduce(
+                        (acc, { id }, index) => ({ ...acc, [id]: snapshotResults[index] }),
+                        {}
+                    );
+                    resolve(result);
+                })
+                .catch((error) => {
+                    reject(error);
+                });
+        });
 
-        try {
-            const snapshots = await Promise.all(
-                configurations.map(({ id }) => fetchLastReportJobForConfiguration(id))
-            );
-            setResult({
-                reportSnapshots: configurations.reduce(
-                    (acc, { id }, index) => ({ ...acc, [id]: snapshots[index] }),
-                    {}
-                ),
-                isLoading: false,
-                error: null,
-            });
-        } catch (error) {
-            setResult({
-                reportSnapshots: {},
-                isLoading: false,
-                error: getErrorMessage(error),
-            });
-        }
+        return promise;
     }, [reportConfigurations]);
+    const { data, isLoading, error, refetch } = useRestQuery(fetchSnapshotsCallback);
 
-    useInterval(fetchSnapshots, 10000);
+    useInterval(refetch, 10000);
 
-    useEffect(() => {
-        // eslint-disable-next-line no-void
-        void fetchSnapshots();
-        // Clear out statuses when report configurations change to avoid stale renders across pages
-        setResult(defaultResult);
-    }, [fetchSnapshots]);
-
-    return {
-        ...result,
-        fetchSnapshots,
+    const result: FetchLastSnapshotReturn = {
+        reportSnapshots: data || {},
+        isLoading,
+        error: getAxiosErrorMessage(error),
```

Comment:
Forgive me if mind is still in Monday mode. Does this need `error &&` as condition?
The way I read the helper function, it would return `Unknown error` here:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/responseErrorUtils.ts#L39

Reply:
I'm confused about what you mean by `error &&`.  Yeah, true it could potentially return the `Uknown error` string. Would it be better if `getAxiosErrorMessage` took an optional second param (or options param for more flexibility) so we can define what the default message should be?

`
error: getAxiosErrorMessage(error, "This is a default error message")
`

Reply:
Promise-based code calls `getAxiosErrorMessage` only in `catch` block when there is an error, but the way I read (maybe misread) this `return` statement is that there might or might not be an error?

Reply:
Ok, I see what you mean. Good catch. We should only call that function on `error` if the error is defined. I'll make sure to fix that.